apps/sharedtest/test/unit/crop_resize_rotate_test.js

```diff
@@ -4,13 +4,123 @@
 /* global Downsample */
-/* global cropResizeRotate */
+/* global cropResizeRotate, JPEGParser */
 
+require('/shared/test/unit/mocks/mock_lazy_loader.js');
 require('/shared/js/blobview.js');
 require('/shared/js/media/jpeg_metadata_parser.js');
 require('/shared/js/media/image_size.js');
 require('/shared/js/media/downsample.js');
 require('/shared/js/media/crop_resize_rotate.js');
+require('/shared/js/media/jpeg-exif.js');
+
+function fetchBlob(url) {
+  return new Promise(function(resolve, reject) {
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', url);
+    xhr.onload = function() {
+      if (xhr.status !== 200) {
+        reject(new Error('Failed with status: ' + xhr.status));
+      } else {
+        resolve(this.response);
+      }
+    };
+    xhr.onerror = xhr.ontimeout = function() {
+      reject(new Error('Failed' + xhr.status));
+    };
+    xhr.responseType = 'blob';
+    xhr.send();
+  });
+}
+
+suite('Metadata copy test', function() {
+  suiteSetup(function() {
+
+    // We need the mock lazy loader for jpeg-exif.
+    window.LazyLoader = window.MockLazyLoader;
+  });
+
+
+  [
+    // this one is horizontal
+    { fileName: '/test-data/blue.jpg', dims: true, rotated: false,
+      targetSize: { width: 640, height: 480 } },
+    { fileName: '/test-data/blue.jpg', dims: true, rotated: false,
+      targetSize: null },
+    { fileName: '/test-data/blue.jpg', dims: true, rotated: false,
+      targetCrop: { left: 10, top: 10, width: 150, height: 100 },
+      targetSize: null },
+    // this one is vertical (Exif orientation)
+    { fileName: '/test-data/red.jpg', dims: true, rotated: true,
+      targetSize: { width: 480, height: 640 } },
+    { fileName: '/test-data/red.jpg', dims: true, rotated: true,
+      targetSize: null },
+    { fileName: '/test-data/red.jpg', dims: true, rotated: true,
+      targetCrop: { left: 10, top: 10, width: 100, height: 150 },
+      targetSize: null }
+  ].
+    forEach(function (testFile) {
+      var testName = 'Testing metadata copy for ' + testFile.fileName;
+      if (testFile.targetSize) {
+        testName += ' resized to ' + testFile.targetSize.width +
+          'x' + testFile.targetSize.height;
+      }
+      test(testName, function(done) {
+
+        fetchBlob(testFile.fileName).then(function (blob) {
+          assert.ok(blob);
+
+          var targetSize = testFile.targetSize;
+          var targetCrop = testFile.targetCrop === undefined ? null :
+              testFile.targetCrop;
+          cropResizeRotate(
+            blob, targetCrop, targetSize, 'image/jpeg+exif',
+            function (error, outBlob) {
+              assert.equal(error, null);
+              assert.ok(outBlob);
+
+              JPEGParser.readExifMetaData(
+                outBlob, function (error, metaData) {
+                  assert.ok(metaData);
+                  // check metaData
+                  assert.equal(metaData.Orientation, 1);
+
+                  if (testFile.dims) {
```

Comment:
dims is true in all 6 of your test cases, so I'm not sure if you really need this...